### PR TITLE
Add allowPauseResponse option to sync webhooks

### PR DESCRIPTION
**Pull request by ravikiranvm**
Part of OPS-4578.

## Additional Notes

Bug: when the child Slack/Teams notification workflow failed before actually sending the message, the parent campaign workflow still marked the opportunity as Assigned. Originating incident: the child's "Format message" code step threw, so the flow never reached the Request Action step — no message was sent, yet the campaign table row was flipped to Assigned (and, being no longer Created, was never retried). Root cause: notify-owners called the child via the async webhook endpoint, which returns 200 on enqueue — so the "Webhook succeeded?" branch passed regardless of what happened in the child run.

Fix — three pieces:

1. `ProgressUpdateType.WEBHOOK_RESPONSE_ON_PAUSE` (`packages/shared`): runs started with it get their synchronous webhook response published as soon as the run pauses, in addition to the existing terminal-state responses (FAILED → 500 etc.). Key insight: the Slack/Teams "Request Action" step sends the message and then pauses the run waiting for user action — so run paused ⇔ message delivered. Publish-on-pause is strictly opt-in: plain `WEBHOOK_RESPONSE` sync calls behave exactly as before (regression-tested), so sync flows that pause on delay steps still receive their final response after resume. The paused response body is the block-provided `pauseMetadata.response` when set, otherwise `{ message: 'The flow is paused' }`.
2. `/v1/webhooks/:flowId/sync?allowPauseResponse=true`: the sync route accepts the opt-in boolean `allowPauseResponse` parameter and starts the run with the new progress-update type. Validated by the route schema: `true`/`false` are accepted (omit the parameter or pass `false` to wait for completion); any other value is rejected with 400 before the run starts.
3. [Will be part of internal PR]Templates notify-owners, remind-owners, escalate-unaddressed-opportunities call the child notification workflow via `/sync?allowPauseResponse=true`. The existing "Webhook succeeded?" (2xx) branches now mean "message delivered (child paused) or flow completed": child fails before sending → 5xx → the row is left untouched and retried next scheduler cycle. remind/escalate had the same defect class (recording Last notified at / Escalated at on enqueue) — same one-line fix, same mechanism.

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

N/A — the only user-visible text change is the Catch Webhook trigger description, which now documents `/sync` and `/sync?allowPauseResponse=true`.

### CODE REVIEW

File: packages/server/api/src/app/webhooks/webhook-controller.ts [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
     SyncWebhookRequest,
-    async (request: FastifyRequest<{ Params: WebhookUrlParams }>, reply) => {
+    async (
+      request: FastifyRequest<{
+        Params: WebhookUrlParams;
+        Querystring: SyncWebhookQueryParams;
+      }>,
+      reply,
+    ) => {
+      const respondOnPause = request.query.respondOnPause ?? false;
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/server/api/src/app/webhooks/webhook-controller.ts]: The route schema declares `respondOnPause` as `Type.Boolean()` and the Fastify AJV config (`server.ts`, `coerceTypes: 'array'`) coerces `"true"`/`"false"` to booleans before the handler runs, so `request.query.respondOnPause` is already a boolean here; `?? false` only covers the omitted-param case. Covered by `webhook-controller.test.ts`: `=false` → handler receives `false`, `=banana` → 400 and the handler is never called.

File: packages/server/api/test/unit/webhooks/webhook-controller.test.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  it('rejects a non-boolean respondOnPause value with 400 without invoking the handler', async () => {
+    const response = await app.inject({
+      method: 'POST',
+      url: '/v1/webhooks/aaaaaaaaaaaaaaaaaaaaa/sync?respondOnPause=banana',
+    });
+
+    expect(response.statusCode).toBe(400);
+    expect(mockHandleWebhook).not.toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/server/api/test/unit/webhooks/webhook-controller.test.ts]: The description was stale, not the test: since the querystring schema was added, invalid values are rejected with 400 by schema validation rather than silently falling back — a typo like `respondOnPause=ture` degrading to the old behaviour is exactly what this change is meant to prevent. Updated the PR description to state the actual contract (`true`/`false` accepted, anything else → 400, omitted → default).

File: packages/server/api/src/app/workers/engine-controller.ts [hunk 3]
```diff
@@ -326,10 +325,32 @@ function getExecutionState(
...
 ): Promise<EngineHttpResponse> {
   switch (result.status) {
+    case FlowRunStatus.PAUSED:
+      return {
+        status: StatusCodes.OK,
+        body: {
+          message: 'The flow is paused',
+        },
+        headers: {},
+      };
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/server/api/src/app/workers/engine-controller.ts:355]: Done in 1f7be736: the paused response body is now `pauseMetadata.response` when a block provides one, falling back to `{ message: 'The flow is paused' }`. For context, nothing in the API/engine consumed `pauseMetadata.response` before and only `wait-for-approval` sets it (to `{}`); Slack/Teams do not set it.

File: packages/server/api/test/unit/engine/engine-controller.test.ts [hunk 4]
```diff
@@ -433,6 +455,92 @@ describe('Engine Controller - update-run endpoint', () => {
...
+
+        expect(webhookResponseWatcher.publish).toHaveBeenCalledWith(
+          'test-run-id',
+          'test-handler-id',
+          {
+            status: StatusCodes.OK,
+            body: { message: 'The flow is paused' },
+            headers: {},
+          },
+        );
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/server/api/test/unit/engine/engine-controller.test.ts:492]: Updated in 1f7be736: the existing test now asserts the flow-provided `pauseMetadata.response` body, and a new case covers the fallback message when the pause metadata has no `response`.

File: packages/server/api/src/app/webhooks/webhook-controller.ts [hunk 5]
```diff
@@ -71,12 +79,23 @@ const WEBHOOK_PARAMS = {
   },
 };
 
+const SyncWebhookQueryParams = Type.Object({
+  waitUntil: Type.Optional(
```
Comment (MarceloRGonc) [inline packages/server/api/src/app/webhooks/webhook-controller.ts]: Change to a boolean 'allowPauseResponse'

Comment (ravikiranvm) [reply packages/server/api/src/app/webhooks/webhook-controller.ts]: Done

**Review by MarceloRGonc — APPROVED** [review]
